File: example/read.cc
#include "person.pb.h"
#include "../recordio.h"

#include <iostream>
#include <string>


int main() {
  // Read
  std::ifstream ifs("example.recordio", std::ios::binary);
  recordio::RecordReader reader(&ifs);

  Person person;
  while(reader.ReadProtocolMessage(&person)) {
    std::cout << person.name() << " (" << person.age() << ")" << std::endl;
  }

  return 0;
}

File: example/read.py
#!/usr/bin/env python2
# -*- coding: utf-8 -*-

from person_pb2 import Person
import recordio

with open("example.recordio", "rb") as f:
    reader = recordio.RecordReader(f)
    person = Person()
    while reader.ReadProtocolMessage(person):
        print person
